lab6/app/src/main/java/com/app/project1/MainActivity.java

```diff
@@ -0,0 +1,237 @@
+    Spinner spinner;
+    String text;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_main);
+        download = findViewById(R.id.download);
+        spinner = findViewById(R.id.spinner);
+        Image = findViewById(R.id.image);
+        download.setOnClickListener(this);
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this, R.array.button,
+                android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+
+        if (ActivityCompat.checkSelfPermission(MainActivity.this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, EXTERNAL_STORAGE_PERMISSION_CONSTANT);
+
+        }
+
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        text = parent.getItemAtPosition(position).toString();
+        if (spinner.getSelectedItem().toString().trim().equals("External")) {
+            text = "External";
+            Toast.makeText(MainActivity.this, "op" + text, Toast.LENGTH_SHORT).show();
+
+        }
+
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> parent) {
+    }
+
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.download:
+                text = String.valueOf(spinner.getSelectedItem());
+
+                if (text.contains("External")) {
+
+                    Picasso.get().load(Link).into(Image);
+                    if (ActivityCompat.checkSelfPermission(MainActivity.this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                        Toast.makeText(this, "No Permission", Toast.LENGTH_SHORT).show();
+                    } else {
+                        MyAsyncTask asyncTask = new MyAsyncTask();
+                        asyncTask.execute();
+                    }
+
+                }
+                    if (text.contains("Internal") || text.contains("Undefined")) {
+
+                        Picasso.get().load(Link).into(Image);
+                        if (ActivityCompat.checkSelfPermission(MainActivity.this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                            Toast.makeText(this, "No Permission", Toast.LENGTH_SHORT).show();
+                        } else {
+                            MYTask asyncTask = new MYTask();
+                            asyncTask.execute();
+                        }
+
+                    }
+                }
+        }
+
+
+
+        @SuppressLint("StaticFieldLeak")
+        class MyAsyncTask extends AsyncTask<Void, Void, Void> {
```

Comment:
We highly discourage the use of AsyncTasks. They will also be deprecated in the next release. Please follow our advice from the labs and use Runnables , Threads instead of AsyncTask:
https://ocw.cs.pub.ro/courses/smd/laboratoare/04